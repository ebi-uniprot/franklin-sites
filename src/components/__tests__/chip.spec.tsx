import { screen, render, fireEvent } from '@testing-library/react';

import Chip from '../chip';

describe('Chip component', () => {
  it('should render', () => {
    const { asFragment } = render(<Chip>Some content</Chip>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should fire onRemove when remove SVG is clicked', () => {
    const onRemove = jest.fn();
    render(<Chip onRemove={onRemove}>Some content</Chip>);
    const removeIcon = screen.getByTestId('remove-icon');
    fireEvent.click(removeIcon);
    expect(onRemove).toHaveBeenCalled();
  });

  it('should not show remove icon when onRemove prop is not passed', () => {
    render(<Chip>Some content</Chip>);
    const removeIcon = screen.queryByTestId('remove-icon');
    expect(removeIcon).toBeNull();
  });

  it('should not have disabled attribute when disabled=false passed as a prop', () => {
    render(<Chip disabled={false}>Some content</Chip>);
    expect(screen.getByRole('button')).not.toHaveClass('chip--disabled');
  });

  it('should have disabled attribute when disabled=true passed as a prop', () => {
    render(<Chip disabled>Some content</Chip>);
    expect(screen.getByRole('button')).toHaveClass('chip--disabled');
  });
});
