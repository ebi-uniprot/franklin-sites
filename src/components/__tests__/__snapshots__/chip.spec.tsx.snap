// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip component should render 1`] = `
<DocumentFragment>
  <span
    class="chip"
    role="button"
    tabindex="0"
  >
    Some content
  </span>
</DocumentFragment>
`;
